R/combine-mrs.R

```diff
@@ -0,0 +1,60 @@
+##' Combine multiple_response variables into a new multiple_response
+##'
+##' This function lets you combine multiple_response variables into a new multiple_response
+##' Arguments include at least two multiple_response variables as well as any metadata
+##' for the new variable that is being created. The argument useNAsubs
+##' is given the defult value FALSE. If it is made TRUE, any subvariables that
+##' are NA in the categorical variables will become subvariables.
+##' This function uses the data from the arrays in the order that they are
+##' given. If the first variable is NA for a given row of a subvariable,
+##' it will use the data from the second variable and so on.
+##' @param variable 1 to combine
+##' @param variable 2 to combine
+##' @param other desired attributes such as new variable alias, name, 
+##' description as well as any other variables you would like to combine
+##' @param a boolean describing whether subvariables that are NA in the
+##' original variables should be made into subvariables in the new
+##' variable
+##' @param a boolean describing whether 'selected' should be given priority
+##' if false, 'not selected' will be given priority
+##' @return a new variable with combined responses from the given variables
+##' @export
+combineMRs <- function(var1, var2, ..., selectedFirst=TRUE){
```

Comment:
If I read the tests for this correctly, the purpose of this function is to take the subvariables from two or more MR variables and make a single MR from them--is that right? If so, how about we keep the first MR, unbind the rest, and use addSubvariables to add them to the first? Like:

```
combineMRs <- function (variable, ...) {
    newsubvars <- unlist(lapply(list(...), unbind)) ## unbind returns the URLs of the unbound variables
    return(addSubvariables(variable, newsubvars))
}
```

addSubvariables would need to take more than just one subvar, and it would need to accept URLs, but neither are too challenging (and they're more generally useful in that function itself too). And also this way, you wouldn't need separate functions to add to categorical arrays and multiple responses--same function would work.